src/test/java/net/testudobank/helpers/MvcControllerIntegTestHelpers.java

```diff
@@ -70,6 +79,19 @@ public static void checkOverdraftLog(Map<String,Object> overdraftLog, LocalDateT
     System.out.println("Timestamp stored in OverdraftLogs table for the Repayment: " + overdraftLogTimestamp);
   }
 
+  // Verifies that a single crypto log in the CryptoHistory table matches the expected customerID, timestamp, action, cryptocurrency, and amount
+  public static void checkCryptoLog(Map<String,Object> cryptoLog, LocalDateTime timeWhenRequestSent, String expectedCustomerID, String expectedAction, String expectedCryptoName, double expectedCryptoAmount) {
+    assertEquals(expectedCustomerID, cryptoLog.get("CustomerID"));
+    assertEquals(expectedAction, cryptoLog.get("Action"));
+    assertEquals(expectedCryptoAmount, ((BigDecimal) cryptoLog.get("CryptoAmount")).doubleValue());
```

Comment:
I have a small concern that the conversion from `BigDecimal` to `double` can sometimes make it not equal the `double expectedCryptoAmount`, but this is probably negligible when considering that this is in a controlled integ test environment, and that we will be using simple CryptoAmount values.

Reply:
Alternatively, I thought of changing everything in the application involving crypto from double to BigDecimal. This is a substantial change though